### PR TITLE
Add targetOwner option to @Check syntax

### PR DESCRIPTION
**Pull request by Drental**
When target:self is defined in the @Checks params, set the target for any defense evaluation to the origin of the Check.

### CODE REVIEW

**Comment by stwlam** [comment]
This shouldn't take a value since there is no other value

**Comment by Drental** [comment]
That does sound reasonable, adjusted the PR accordingly

File: src/module/system/text-editor.ts [hunk 1]
```diff
@@ -695,6 +696,7 @@ class TextEditorPF2e extends TextEditor {
                 pf2Label: localize("DCWithName", { name }),
                 pf2Adjustment: Number(params.adjustment) || null,
                 pf2Roller: params.roller || null,
+                pf2TargetSelf: params.targetSelf,
```
Comment (stwlam) [inline src/module/system/text-editor.ts:699]: no `pf2`, and make it `targetOwner` so that there's no confusion of perspective.

Comment (Drental) [reply src/module/system/text-editor.ts]: targetOwner is a much better name! Thanks!

File: src/module/system/text-editor.ts [hunk 2]
```diff
@@ -583,6 +583,7 @@ class TextEditorPF2e extends TextEditor {
                 ...(rawParams.options?.split(",").map((t) => t.trim()) ?? []),
             ]).sort(),
         };
+        if ("targetSelf" in rawParams) params.targetOwner = true;
```
Comment (stwlam) [inline src/module/system/text-editor.ts:586]:
```suggestion
        if ("targetOwner" in rawParams) params.targetOwner = true;
```

File: src/module/system/text-editor.ts [hunk 3]
```diff
@@ -1074,6 +1076,7 @@ interface CheckLinkParams {
     /** Refrain from adding domains to the check. */
     immutable: boolean;
     roller?: string;
+    targetOwner?: boolean;
```
Comment (stwlam) [inline src/module/system/text-editor.ts:1079]:
```suggestion
    targetOwner: boolean;
```

File: src/module/system/text-editor.ts [hunk 4]
```diff
@@ -697,6 +698,7 @@ class TextEditorPF2e extends TextEditor {
                 pf2Label: localize("DCWithName", { name }),
                 pf2Adjustment: Number(params.adjustment) || null,
                 pf2Roller: params.roller || null,
+                TargetOwner: params.targetOwner,
```
Comment (stwlam) [inline src/module/system/text-editor.ts:701]:
```suggestion
                targetOwner: params.targetOwner,
```

File: src/module/system/text-editor.ts [hunk 5]
```diff
@@ -583,6 +583,7 @@ class TextEditorPF2e extends TextEditor {
                 ...(rawParams.options?.split(",").map((t) => t.trim()) ?? []),
             ]).sort(),
         };
```
Comment (stwlam) [inline src/module/system/text-editor.ts:585]:
```suggestion
    targetOwner: "targetOwner" in rawParams,
        };
```

File: src/module/system/text-editor.ts [hunk 6]
```diff
@@ -583,6 +583,7 @@ class TextEditorPF2e extends TextEditor {
...
+        params.targetOwner = "targetOwner" in rawParams;
```
Comment (stwlam) [inline src/module/system/text-editor.ts:586]:
```suggestion
```

File: src/module/system/text-editor.ts [hunk 7]
```diff
@@ -582,6 +582,7 @@ class TextEditorPF2e extends TextEditor {
                 ...(basic ? ["damaging-effect"] : []),
                 ...(rawParams.options?.split(",").map((t) => t.trim()) ?? []),
             ]).sort(),
+    targetOwner: "targetOwner" in rawParams,
```
Comment (stwlam) [inline src/module/system/text-editor.ts:585]:
```suggestion
            ]).sort(),
            targetOwner: "targetOwner" in rawParams,
```

**Pull request merged**